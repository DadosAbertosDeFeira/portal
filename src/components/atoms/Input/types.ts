import type { ComponentPropsWithRef } from "react";

export type InputVariants = "outline" | "default" | "shadowed";

export type InputProps = Omit<ComponentPropsWithRef<"input">, "prefix"> & {
  name: string;
  label: string;
  suffix?: React.ReactElement;
  hideLabel?: boolean;
  prefix?: React.ReactElement | false;
  errorText?: string;
  variant?: InputVariants;
  containerProps?: ComponentPropsWithRef<"label">;
  helperText?: string;
  filled?: boolean;
};
