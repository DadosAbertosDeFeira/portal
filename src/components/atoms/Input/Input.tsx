import classNames from "classnames";
import type { ComponentPropsWithRef } from "react";
import React, { forwardRef } from "react";

export type InputProps = {} & ComponentPropsWithRef<"input">;

export const Input = forwardRef<HTMLInputElement, InputProps>(function Input(
  { className, ...props },
  ref
) {
  return (
    <input
      {...props}
      className={classNames(
        "w-full rounded border border-gray-2 bg-white p-2 pl-4 font-medium text-gray-dark",
        className
      )}
      {...props}
      ref={ref}
    />
  );
});
