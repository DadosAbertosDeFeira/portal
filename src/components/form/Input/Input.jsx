/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import { useController } from 'react-hook-form';

function Input({ name, label, control, rules }) {
  const { field: input, fieldState: state } = useController({
    name,
    control,
    rules,
    defaultValue: '',
  });

  return (
    <label htmlFor={name}>
      <input
        className="input-field"
        placeholder={label}
        aria-errormessage={state.error?.message}
        aria-invalid={state.invalid}
        aria-label={label}
        {...input}
      />

      {state.error && (
        <span
          key={state.error}
          className="text-alert text-sm anima-input-error-message"
        >
          {state.error.message}
        </span>
      )}
    </label>
  );
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  control: PropTypes.objectOf(PropTypes.any).isRequired,
  rules: PropTypes.objectOf(PropTypes.any),
};

Input.defaultProps = {
  rules: {},
};

export default Input;
