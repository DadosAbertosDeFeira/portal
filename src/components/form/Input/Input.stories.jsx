/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { useForm } from "react-hook-form";
import { Input } from "./Input";

export default {
  title: "components/Input",
  component: Input,
  argTypes: {
    control: {
      control: false,
    },
    name: {
      control: false,
    },
  },
};

export const Default = (args) => {
  const form = useForm();

  return <Input {...args} control={form.control} />;
};

export const Invalid = (args) => {
  const form = useForm();

  form.setError(args.name, {
    type: "required",
    message: "Esse valor é obrigatório.",
  });

  form.setValue(args.name, " ");

  return <Input {...args} control={form.control} rules={{ required: true }} />;
};

Invalid.args = {
  name: "Email",
  label: "E-mail",
};
