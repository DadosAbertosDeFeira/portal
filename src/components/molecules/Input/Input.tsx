import type { Control } from "react-hook-form";
import { useController } from "react-hook-form";

export interface InputProps {
  name: string;
  label?: string;
  defaultValue?: string;
  control: Control<any>;
  rules: Partial<{
    validate: {
      [k: string]: (v: string) => string | boolean;
    };
    required: boolean | string;
  }>;
}

export function Input({
  name,
  control,
  label,
  rules,
  defaultValue = "",
  ...props
}: InputProps) {
  const { field: input, fieldState: state } = useController({
    name,
    control,
    rules,
    defaultValue,
  });

  return (
    <label htmlFor={name}>
      <input
        className="w-full rounded border border-gray-2 bg-white p-2 pl-4 font-medium text-gray-dark"
        placeholder={label}
        aria-errormessage={state.error?.message}
        aria-labelledby={`${name}-input--error-message`}
        aria-invalid={state.invalid}
        aria-label={label}
        {...props}
        {...input}
      />

      {state.error && (
        <span
          key={state.error.message}
          id={`${name}-input--error-message`}
          className="anima-input-error-message text-sm text-alert"
        >
          {state.error.message}
        </span>
      )}
    </label>
  );
}
