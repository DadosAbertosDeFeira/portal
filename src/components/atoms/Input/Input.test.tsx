import { render } from "@testing-library/react";

import { Input, type InputProps } from "./Input";

describe("Input", () => {
  const makeSut = ({ ...props }: Partial<InputProps>) => {
    return render(<Input {...props} />);
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment).toMatchSnapshot();
  });
});
