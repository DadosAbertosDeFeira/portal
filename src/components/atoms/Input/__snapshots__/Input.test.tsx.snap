// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `[Function]`;
